integration_tests/Containerfile

```diff
@@ -0,0 +1,82 @@
+# Use the base Debian image from the Podman hub
+FROM docker.io/library/debian
+
+
+
+# INSTALLING DEPENDENCIES
+
+# Update package repositories and install necessary packages
+RUN apt update -y
+
+# Copy and install dependencies before setup to save build time 
+COPY ../documentation/dependencies/debian.dependencies /dependencies/debian.dependencies
+COPY ./integration_tests/dependencies/kernel.dependencies /dependencies/kernel.dependencies
+
+RUN DEBIAN_DEPENDENCIES=$(cat /dependencies/debian.dependencies) \
+    && apt install -y $DEBIAN_DEPENDENCIES
+
+RUN KERNEL_DEPENDENCIES=$(cat /dependencies/kernel.dependencies) \
+    && apt install -y $KERNEL_DEPENDENCIES
+
+
+
+# CREATING USER
+
+# Create a new user
+RUN useradd -ms /bin/bash user
+
+# Switch to user
+USER user
+
+# Set the working directory to the user's home directory
+WORKDIR /home/user
+
+
+
+# CLONE KERNEL TREE
+
+# TODO: Is cached. What if kernel is updated?
+RUN git clone --depth 1 https://github.com/torvalds/linux.git
```

Comment:
Can we add a logic like:
1. If there is no repo, clone it.
2. If there is a repo, update it.